pkg/controllerhelpers/scylla.go

```diff
@@ -15,46 +16,110 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/errors"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	corev1schedulinghelpers "k8s.io/component-helpers/scheduling/corev1"
 	"k8s.io/component-helpers/scheduling/corev1/nodeaffinity"
 )
 
-func GetScyllaIPFromService(svc *corev1.Service) (string, error) {
-	if svc.Spec.Type != corev1.ServiceTypeClusterIP {
-		return "", fmt.Errorf("service %s is of type %q instead of %q", naming.ObjRef(svc), svc.Spec.Type, corev1.ServiceTypeClusterIP)
+func GetScyllaHost(ctx context.Context, sc *scyllav1.ScyllaCluster, coreV1Client corev1client.CoreV1Interface, svc *corev1.Service) (string, error) {
+	if sc.Spec.ExposeOptions == nil || sc.Spec.ExposeOptions.BroadcastOptions == nil {
+		host, err := getServiceClusterIP(svc)
+		if err != nil {
+			return "", fmt.Errorf("can't get ClusterIP from service %q: %w", naming.ObjRef(svc), err)
+		}
+
+		return host, nil
+	}
+
+	switch sc.Spec.ExposeOptions.BroadcastOptions.Nodes.Type {
+	case scyllav1.BroadcastAddressTypeServiceClusterIP:
+		host, err := getServiceClusterIP(svc)
+		if err != nil {
+			return "", fmt.Errorf("can't get ClusterIP from service %q: %w", naming.ObjRef(svc), err)
+		}
+
+		return host, nil
+	case scyllav1.BroadcastAddressTypePodIP:
+		// Assume API's default
+		podIPSource := scyllav1.StatusPodIPSource
+		if sc.Spec.ExposeOptions.BroadcastOptions.Nodes.PodIP != nil {
+			podIPSource = sc.Spec.ExposeOptions.BroadcastOptions.Nodes.PodIP.Source
+		}
+		switch podIPSource {
+		case scyllav1.StatusPodIPSource:
+			pod, err := coreV1Client.Pods(sc.Namespace).Get(ctx, naming.PodNameFromService(svc), metav1.GetOptions{})
+			if err != nil {
+				return "", fmt.Errorf("can't get pod for service %q: %w", naming.ObjRef(svc), err)
+			}
+
+			host, err := getPodStatusPodIP(pod)
+			if err != nil {
+				return "", fmt.Errorf("can't get PodIP from pod %q status: %w", naming.ObjRef(pod), err)
+			}
+
+			return host, nil
```

Comment:
this is similar to what we do to create member config and it should likely live in a shared function